<?xml version="1.0"?>
<!--
  Launch calculation of individual magnetic vectors
-->
<launch>
  


<!-- run node -->
  <node name="magnetic_field_vector1" pkg="magnetic_field_localization" type="magnetic_field_vector_fast" output="screen">
    <param name="magnetometer_topic" type="str" value="/magnetic1" />
    <param name="vector_topic" type="str" value="/magnetic_vector0" />
    <param name="magnetometer_frame" type="str" value="/magnetometer0" />
    <param name="magnetic_field_frequency" type="int" value="50" />
    <param name="number_of_points_for_analysis" type="int" value="40" />
    <param name="number_of_cycles_for_analysis" type="int" value="1" />
    <param name="magnetometer_sampling_time" type="double" value="0.002" />
    <param name="scaling" type="double" value="0.0146" />
  </node>

  <node name="magnetic_field_vector2" pkg="magnetic_field_localization" type="magnetic_field_vector_fast" output="screen">
    <param name="magnetometer_topic" type="str" value="/magnetic2" />
    <param name="vector_topic" type="str" value="/magnetic_vector1" />
    <param name="magnetometer_frame" type="str" value="/magnetometer1" />
    <param name="magnetic_field_frequency" type="int" value="50" />
    <param name="number_of_points_for_analysis" type="int" value="40" />
    <param name="number_of_cycles_for_analysis" type="int" value="1" />
    <param name="magnetometer_sampling_time" type="double" value="0.002" />
    <param name="scaling" type="double" value="0.0146" />
  </node>
  <node name="magnetic_field_vector3" pkg="magnetic_field_localization" type="magnetic_field_vector_fast" output="screen">
    <param name="magnetometer_topic" type="str" value="/magnetic4" />
    <param name="vector_topic" type="str" value="/magnetic_vector2" />
    <param name="magnetometer_frame" type="str" value="/magnetometer2" />
    <param name="magnetic_field_frequency" type="int" value="50" />
    <param name="number_of_points_for_analysis" type="int" value="40" />
    <param name="number_of_cycles_for_analysis" type="int" value="1" />
    <param name="magnetometer_sampling_time" type="double" value="0.002" />
    <param name="scaling" type="double" value="0.0146" />
  </node>

  <node pkg="tf" type="static_transform_publisher" name="magnetometer_position0" args="0  0.058 -0.123 0 0 0 /base_link /imu1 100"/> 
  <node pkg="tf" type="static_transform_publisher" name="magnetometer_position1" args="0 -0.168 -0.123 3.14159 0 0 /base_link /imu2 100"/> 

  <!--node pkg="tf" type="static_transform_publisher" name="magnetometer_position2" args="0 0.19 -0.355 0 0 0 /base_link /imu3 100"/--> 
  <node pkg="tf" type="static_transform_publisher" name="magnetometer_position2" args="0 -0.3 -0.355 0 0 0 /base_link /imu3 100"/> 

</launch>
